NumberTheory/PerfectNumbers.cpp

```diff
@@ -0,0 +1,37 @@
+ /*
+    Perfect number:
+    prints up to N perfect numbers (a positive integer that 
+    is equal to the sum of its proper positive divisors)
+    Time complexity:
+    O(sqrt(N)), where N is the number being checked 
+    Space complexity:
+    O(1), constant amount of space needed
+*/
+
+#include <iostream>
+
+using namespace std;
+
+bool is_perfect(int num) {
+	int sum = 1;
+	for(int divisor = 2; divisor <= num/2; divisor++)
+	{
+		if(num % divisor == 0)
+			sum += divisor;
+	}
+	if(divisor == sum)
+		return 1;
+	return 0;
+}
+
+int main() {
+	int num;
+	cout << "The number of perfect numbers you want printed: ";
+	cin >> num;

```

Comment:
Instead of asking for
> the number of perfect numbers to be printed

ask for
> the number upto which perfect numbers are to be found

as there are only 8 perfect numbers between 1 and 2<sup>64</sup>-1.

You would also need to change `num` to `limit` or something more meaningful, and also it's type to `unsigned long long` to support a higher range of values.